jina/peapods/runtimes/zmq/zed.py

```diff
@@ -251,24 +251,33 @@ def _handle(self) -> 'ZEDRuntime':
     def _handle_control_req(self):
         # migrated from previous ControlDriver logic
         if self.request.command == 'TERMINATE':
+            self.logger.info('DO TERMINATE')
             self.envelope.status.code = jina_pb2.StatusProto.SUCCESS
             raise RuntimeTerminated
         elif self.request.command == 'STATUS':
             self.envelope.status.code = jina_pb2.StatusProto.READY
             self.request.parameters = vars(self.args)
         elif self.request.command == 'IDLE':
+            self.logger.info('DO IDLE')
             self._idle_dealer_ids.add(self.envelope.receiver_id)
             self._zmqstreamlet.resume_pollin()
             self.logger.debug(
                 f'{self.envelope.receiver_id} is idle, now I know these idle peas {self._idle_dealer_ids}'
             )
+            self.logger.info('DONE IDLE')
         elif self.request.command == 'CANCEL':
+            self.logger.info('DO CANCEL')
             if self.envelope.receiver_id in self._idle_dealer_ids:
                 self._idle_dealer_ids.remove(self.envelope.receiver_id)
+            self.logger.info('DONE CANCEL')
         elif self.request.command == 'ACTIVATE':
+            self.logger.info('DO ACTIVATE')
             self._zmqstreamlet._send_idle_to_router()
+            self.logger.info('DONE ACTIVATE')
```

Comment:
just as warning this is confusing, these `send_idle_to_router` and `Send_cancel_to_router` I do not think wait for a response from `router`